### PR TITLE
delete requires the password the instance was required with

### PR DESCRIPTION
**Pull request by byjott**
The password is given as 'db_pwd' in json to send with the DELETE request. It is verified by logging in with the db_user into this instance, so the current postgres password should be used.

### CODE REVIEW

File: postgraas_server/postgres_instance_driver.py [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-import hashlib
 import psycopg2
 import time
+import socket
 
 
-def get_unique_id(connection_dict):
-    return hashlib.sha1(str(frozenset(connection_dict.items()))).hexdigest()
+def get_hostname():
+    return socket.getfqdn()
```
Comment (sebastianneubauer) [inline postgraas_server/postgres_instance_driver.py]: mmh, I thought about that one too, but it is not working in general I guess, quite often, the instance will be somehow rerouted, at least, the hostname FQDN must not be the hostname we pass to the user. So maybe remove this one here and maybe make another PR...

File: postgraas_server/postgres_instance_driver.py [hunk 2]
```diff
@@ -1,21 +1,20 @@
 __author__ = 'sebastian neubauer'
 import docker
-import hashlib
 import psycopg2
 import time
+import socket
 
 
-def get_unique_id(connection_dict):
```
Comment (sebastianneubauer) [inline postgraas_server/postgres_instance_driver.py]: good catch, nowhere used, isn't it?

File: postgraas_server/postgres_instance_driver.py [hunk 3]
```diff
@@ -26,14 +25,12 @@ def get_open_port():
 
 
 def get_container_by_name(postgraas_instance_name):
-    c = docker.Client(base_url='unix://var/run/docker.sock',
-                        version='auto',
-                        timeout=30)
-    containers = c.containers()
-    for container in containers:
-        for name in container['Names']:
-            if postgraas_instance_name in name.replace("/", ""):
```
Comment (sebastianneubauer) [inline postgraas_server/postgres_instance_driver.py]: is this fixed now? before this workaround was needed...

File: postgraas_server/postgres_instance_driver.py [hunk 4]
```diff
@@ -55,29 +50,51 @@ def create_postgres_instance(postgraas_instance_name, connection_dict):
...
-    c.start(container_id)
-    return container_id
+    container = c.containers.create(image,
+                                    name=postgraas_instance_name,
+                                    ports={internal_port: connection_dict['port']},
+                                    environment=environment,
+                                    restart_policy={"Name": "unless-stopped"},
+                                    labels={"postgraas": image})
+    container.start()
+    return container.id
```
Comment (sebastianneubauer) [inline postgraas_server/postgres_instance_driver.py]: nice!

File: postgraas_server/postgres_instance_driver.py [hunk 5]
```diff
@@ -55,29 +50,51 @@ def create_postgres_instance(postgraas_instance_name, connection_dict):
...
+    for i in range(max(int(timeout * 10), 1)):
+        # not very efficient to always gather all logs, but this method
+        # is currently used for testing only, so it should be OK.
+        output = cont.logs(stdout=True, stderr=True)
+        # the startup script in the docker image starts postgres twice,
+        # so wait for the second start:
+        if output.count('database system is ready to accept connections') >= 2:
+            return True
+        time.sleep(0.1)
+    return False   # pragma: no cover
```
Comment (sebastianneubauer) [inline postgraas_server/postgres_instance_driver.py]: Why is this needed, isn't the `wait_for_postgres` function sufficient?

File: tests/test_integration/test_create_db.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    # test the standalone 'create_db' and 'init_db' scripts.
+    if pid.check_container_exists('postgraas_master_db'):
+        pid.delete_postgres_instance('postgraas_master_db')
+    create_db_file = create_db.__file__
+    cov_runner(create_db_file)
+    # also check the path where the container already exists:
+    cov_runner(create_db_file)
+
+    init_db_file = init_db.__file__
+    cov_runner(init_db_file)
```
Comment (sebastianneubauer) [inline tests/test_integration/test_create_db.py]: this one breaks the tests on travis, any ideas?

File: tests/test_integration/test_postgras_api.py [hunk 7]
```diff
@@ -144,7 +217,7 @@ def test_return_postgres_instance_api(self):
             u'db_name': u'test_return_postgres_instance',
             u'username': u'db_user',
             u'password': u'',
-            u'hostname': u'not imlemented yet',
+            u'hostname': socket.getfqdn(),
```
Comment (sebastianneubauer) [inline tests/test_integration/test_postgras_api.py]: see above, I would remove this feature for now, this doesn't work in general...

File: postgraas_server/postgres_instance_driver.py (hunk 1 above)
Comment (byjott) [inline postgraas_server/postgres_instance_driver.py]: Yes, you're right this is for another PR. Will revert this change in the next commit.

File: postgraas_server/postgres_instance_driver.py (hunk 2 above)
Comment (byjott) [inline postgraas_server/postgres_instance_driver.py]: Right, it's not used anywhere.

File: postgraas_server/postgres_instance_driver.py (hunk 3 above)
Comment (byjott) [inline postgraas_server/postgres_instance_driver.py]: Yes, this is now much cleaner with the docker 2 python interface

File: postgraas_server/postgres_instance_driver.py (hunk 5 above)
Comment (byjott) [inline postgraas_server/postgres_instance_driver.py]: The postgres docker image
1. starts postgres
2. creates the role according to the env variables
3. restarts postgres (dropping any connections)

My first attempt was to just wait until a connection to the db can be made, but in the tests, that would almost always end up making the connections just after 1., which would lead to dropped connections.
I did not want to use `wait_for_postgres` as this could mean waiting for 9 minutes, which I think is too long for a test to fail.
OTOH, this is only used in text code, so I'll move this to `tests`, which I hope makes it more acceptable, given how hacky it is.

File: postgraas_server/postgres_instance_driver.py (hunk 3 above)
Comment (sebastianneubauer) [inline postgraas_server/postgres_instance_driver.py]: cool
